### PR TITLE
Change the Issuance Authorization Signature scheme to the BIP340 Schnorr scheme, and rename `asset_id` to `asset_base`

### PR DESCRIPTION
**Pull request by vivek-arte**
This PR adds the BIP340 python reference implementation to the project files, and imports from it. 

It then generates Schnorr signing and verifying keys as per the BIP 340 implementation, and generates signatures for random 32 byte messages, and renders the output into Rust test vectors for use in the orchard crate.

### CODE REVIEW

File: test-vectors/rust/zsa_issuance_auth_sig.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
+// From https://github.com/zcash-hackworks/zcash-test-vectors/ (zsa_issuance_auth_sig)
```
Comment (PaulLaux) [inline test-vectors/rust/zsa_issuance_auth_sig.rs:1]: remove `zsa_` prefix from file name.

File: zcash_test_vectors/orchard/zsa_issuance_auth_sig.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
```
Comment (PaulLaux) [inline zcash_test_vectors/orchard/zsa_issuance_auth_sig.py:1]: remove `zsa_` prefix from file name

**Review by PaulLaux — DISMISSED** [review]
Approved pending the required changes

File: test-vectors/rust/zsa_issuance_auth_sig.rs (hunk 1 above)
Comment (vivek-arte) [reply test-vectors/rust/zsa_issuance_auth_sig.rs:1]: Done

File: zcash_test_vectors/orchard/zsa_issuance_auth_sig.py (hunk 2 above)
Comment (vivek-arte) [reply zcash_test_vectors/orchard/zsa_issuance_auth_sig.py:1]: Done

File: test-vectors/rust/issuance_auth_sig.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// From https://github.com/zcash-hackworks/zcash-test-vectors/ (zsa_issuance_auth_sig)
+// From https://github.com/QED-it/zcash-test-vectors/ (issuance_auth_sig)
```
Comment (PaulLaux) [inline test-vectors/rust/issuance_auth_sig.rs:1]: manual changes in `.rs` files are not going to persist a regeneration, revert this change.

**Review by PaulLaux — APPROVED** [review]
approved with one comment

File: test-vectors/rust/issuance_auth_sig.rs (hunk 3 above)
Comment (PaulLaux) [reply test-vectors/rust/issuance_auth_sig.rs:1]: Revert to `// From https://github.com/zcash-hackworks/zcash-test-vectors/ (issuance_auth_sig)`

**Review by PaulLaux — APPROVED** [review]
